### ISSUE TITLE
Chunked transfert type only transfers partial data

### ISSUE THREAD

**Issue by BlackYoup** [sozu-proxy/sozu#194]
Hi,

I've come across this issue while streaming files. The target server streams a file using chunked encoding. When using sozu, I only get ~12KB of data out of a 1MB file.

Here is a reproduction case:

Server.js:
```javascript
const fs = require("fs");
const path = require("path");
const server = require("express")();

const PORT = process.env.PORT || 3000;

server.get('/', (req, res) => {
  const binPath = path.resolve("./test");
  fs.access(binPath, fs.constants.R_OK, err => {
      console.log(`Serving file ${binPath} to client`);
      res.status(200);
      res.setHeader("content-type", "application/x-tar");
      fs.createReadStream(binPath).pipe(res);
  });
});

server.listen(PORT, () => {
  console.log(`Listening on port ${PORT}`);
});
```

The `./test` file is a 1MB file. Just curl `/` to try to download the file.

Sozu logs (nothing's after):
```
2017-07-23T21:28:50Z    528097019449187 20420   DEBUG   TAG-0   sozu_lib::network::protocol::http      b7ba0c75-757c-4db9-a1cb-33f4f4d49b88    unknown       FRONT [Some(Token(4))]: read 80 bytes
2017-07-23T21:28:50Z    528097019705704 20420   TRACE   TAG-0   sozu_lib::network::session     front readable  interpreting client order ConnectBackend
2017-07-23T21:28:50Z    528097019838103 20420   INFO    TAG-0   b7ba0c75-757c-4db9-a1cb-33f4f4d49b88   unknown         Connecting MyApp -> (V4(127.0.0.1:1026), 1, 0)
2017-07-23T21:28:50Z    528097020312121 20420   DEBUG   TAG-0   sozu_lib::network::protocol::http      b7ba0c75-757c-4db9-a1cb-33f4f4d49b88    unknown       BACK [4->504]: wrote 233 bytes of 233
2017-07-23T21:28:50Z    528097021678329 20420   DEBUG   TAG-0   sozu_lib::network::protocol::http      b7ba0c75-757c-4db9-a1cb-33f4f4d49b88    unknown       BACK  [4<-504]: read 16384 bytes
2017-07-23T21:28:50Z    528097022024014 20420   TRACE   TAG-0   sozu_lib::parser::http11       PARSER  got Connection header: ""keep-alive""
2017-07-23T21:28:50Z    528097022135245 20420   DEBUG   TAG-0   sozu_lib::parser::http11       PARSER  headers parsed, stopping
2017-07-23T21:28:50Z    528097022379400 20420   TRACE   TAG-0   sozu_lib::network::session     front writable  interpreting client order Continue
2017-07-23T21:28:50Z    528097022443811 20420   DEBUG   TAG-0   sozu_lib::network::protocol::http      b7ba0c75-757c-4db9-a1cb-33f4f4d49b88    unknown       BACK  [4<-504]: read 16384 bytes
```

Tcpdump shows that the server sends a chunk but nothing more after, just a few ack.

**Comment by Geal** [comment]
it should be fixed with b26c111. That one was not easy to debug :/

**Comment by BlackYoup** [comment]
Works great !
